UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/HealthCheck/EventHubLoggingHealthClient.cs

```diff
@@ -0,0 +1,46 @@
+﻿
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Producer;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using UKHO.ExternalNotificationService.Common.Configuration;
+using UKHO.ExternalNotificationService.Common.Logging;
+
+namespace UKHO.ExternalNotificationService.Common.HealthCheck
+{
+    [ExcludeFromCodeCoverage]
+    public class EventHubLoggingHealthClient : IEventHubLoggingHealthClient
+    {
+        private readonly IOptions<EventHubLoggingConfiguration> _eventHubLoggingConfiguration;
+
+        public EventHubLoggingHealthClient(IOptions<EventHubLoggingConfiguration> eventHubLoggingConfiguration)
+        {
+            this._eventHubLoggingConfiguration = eventHubLoggingConfiguration;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            var eventHubProducerClient = new EventHubProducerClient(_eventHubLoggingConfiguration.Value.ConnectionString, _eventHubLoggingConfiguration.Value.EntityPath);
+            using EventDataBatch eventBatch = await eventHubProducerClient.CreateBatchAsync();
```

Comment:
`CreateBatchAsync` has an overload with cancellation support so please use it:
```suggestion
            using EventDataBatch eventBatch = await eventHubProducerClient.CreateBatchAsync(cancellationToken);
```

Reply:
Yes, I have passed cancellationToken as parameter